2DRove/Assets/Scripts/StateMachines/RockRobotBossFSM/RockBossAttackState.cs

```diff
@@ -0,0 +1,114 @@
+using System.Collections;
+using UnityEngine;
+
+public class RockBossAttackState : RockBossBaseState
+{
+    private float attackTime = 2.5f;
+    private Animator animator;
+    private int returnAttackType;
+    public override void EnterState(RockBossStateManager RockBoss)
+    {
+        Debug.Log("Entering Attack State");
+        attackTime = 2.5f;
+        animator = RockBoss.GetComponent<Animator>();
+        PerformRandomAttack(RockBoss);
+    }
+
+    private void PerformRandomAttack(RockBossStateManager RockBoss)
+    {
+        int attackType = Random.Range(0, 4);
+        Debug.Log(attackType);
+        switch (attackType)
+        {
+            case 0:
+                animator.SetBool("Beaming", true);
+                Debug.Log("Beaming");
+                break;
+            case 1:
+                animator.SetBool("Shooting", true);
+                Debug.Log("Shooting");
+                break;
+            case 2:
+                animator.SetBool("Charging", true);
+                Debug.Log("Charging");
+                break;
+            case 3:
+                animator.SetBool("Slamming", true);
+                Debug.Log("Slamming");
+                break; 
+        }
+        returnAttackType = attackType;
+    }
+
+    public override void UpdateState(RockBossStateManager RockBoss)
+    {
+        if(attackTime <= 0)
+        {
+            animator.SetBool("Beaming", false);
+            animator.SetBool("Shooting", false);
+            animator.SetBool("Charging", false);
+            animator.SetBool("Spinning", false);
```

Comment:
Remove this line, you took this bool out already.

Reply:
noted, will change very soon